Lib/test/test_asyncio/test_base_events.py

```diff
@@ -834,19 +835,21 @@ async def test():
 
     def test_create_named_task_with_custom_factory(self):
         def task_factory(loop, coro):
-            return asyncio.Task(coro, loop=loop)
+            assert False
 
         async def test():
             pass
 
-        loop = asyncio.new_event_loop()
-        loop.set_task_factory(task_factory)
-        task = loop.create_task(test(), name='test_task')
-        try:
-            self.assertEqual(task.get_name(), 'test_task')
-        finally:
-            loop.run_until_complete(task)
-            loop.close()
+        with (
+            contextlib.closing(asyncio.EventLoop()) as loop,
+            contextlib.closing(test()) as coro,
+        ):
+            loop.set_task_factory(task_factory)
```

Comment:
Rather than this, change it to take name arg in factory and keep the test intact